compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5604,6 +5625,46 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
         }
     }
 
+    private BType getTypeOfTypeParameter(BType selectType, Location pos) {
+        BType referredType = Types.getReferredType(selectType);
+        if (referredType.tag == TypeTags.INTERSECTION) {
+            referredType = ((BIntersectionType) referredType).effectiveType;
+        }
+
+        if (referredType.tag == TypeTags.UNION) {
+            BUnionType unionType = (BUnionType) referredType;
+            LinkedHashSet<BType> memberTypes = new LinkedHashSet<>();
+            for (BType type : unionType.getMemberTypes()) {
+                BType mapType = getTypeOfTypeParameter(type, pos);
+                if (mapType == symTable.semanticError) {
+                    return symTable.semanticError;
+                }
+                memberTypes.add(mapType);
+            }
+            return new BUnionType(null, memberTypes, false, false);
```

Comment:
Cannot we add a test which comes this line?

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/36352/commits/302e0c7adcc79a8cbc6e1eb2090b1dabd8b896d5.